mongo/datadog_checks/mongo/collectors/custom_queries.py

```diff
@@ -9,6 +13,34 @@
     ReplicaSetDeployment,
 )
 
+MONGO_DATE_EXPRESSIONS = {
+    r"ISODate\(\'(.*?)\'\)": (lambda m: dateutil.parser.isoparse(m.groups()[0])),
+    r"ISODate\(\)|Date\(\)": (lambda m: datetime.datetime.now(tz=tzutc())),
+    r"new\s*Date\(ISODate\(\).getTime\(\)\s*-\s*(\d+)": (
```

Comment:
Note that this is very very fragile, i.e. it can fail in ways that are confusing for users, such as:

```
ISODate(  )
new Date(ISODate().getTime() - 1000 * 3600)
(Date())
...
```

I see how this can provide immediate relief for some specific use cases, but this might open a can of worms as there may not be going back (to avoid breaking) and we may create false expectations if we start to support subsets of JavaScript.

Reply:
Improved regular expressions and updated spec.yaml with some examples